File: android/src/main/java/br/dev/michaellopes/flutter_anycam/camera/CameraFactory.java
package br.dev.michaellopes.flutter_anycam.camera;

import java.util.Map;

import br.dev.michaellopes.flutter_anycam.model.ViewCameraSelector;

public class CameraFactory {

    private static final CameraFactory instance = new CameraFactory();

    private CameraFactory() {}

    public static CameraFactory getInstance() {
        return instance;
    }

    public BaseCamera getCamera(int vId, Map<String, Object> params) {
        synchronized (instance) {
            Map<String, Object> map = (Map<String, Object>) params.get("cameraSelector");
            ViewCameraSelector cameraSelector = ViewCameraSelector.fromMap(map);
            switch (cameraSelector.getLensFacing()) {
                case "usb":
                    return new UsbCamera(vId, params);
                case "rtsp":
                    return new RTSPCamera(vId, params);
                case "back":
                case "front":
                default:
                    return new DeviceCamera(vId, params);
            }
        }
    }
}

File: ios/Classes/Camera/CameraFactory.swift
//
//  CameraFactory.swift
//  Pods
//
//  Created by Michael Lopes on 08/08/25.
//
class CameraFactory {
    //as [String: Any?] ?? [String: Any?]()
    let frame: CGRect;
    var params: [String : Any?] = [String: Any?]();
    
    init(frame: CGRect) {
       self.frame = frame
    }
    
    func getCamera(vId: Int, args: Any?) -> BaseCamera{
    
        self.params = (args as? [String: Any?]) ?? [String: Any?]();
        let map = params["cameraSelector"] as? [String : Any?];
        let cameraSelector = ViewCameraSelector.fromMap(map!);
                
        switch cameraSelector?.lensFacing {
        case "rtsp": 
            return RTSPCamera(frame: self.frame, viewId: vId, params: self.params)
        case "back":
            return DeviceCamera(frame: self.frame, viewId: vId, params: self.params)
        case "front":
            return DeviceCamera(frame: self.frame, viewId: vId, params: self.params)
        default:
            return DeviceCamera(frame: self.frame, viewId: vId, params: self.params)
        }
    }
}
